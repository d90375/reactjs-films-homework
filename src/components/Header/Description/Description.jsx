import React from "react";
import "./Description.scss";
import Rating from "./Rating";

const Description = () => {
  return (
    <div className="description">
      <span className="description__title">The Jungle Book</span>
      <div className="description__specification">
        <span className="specification__genre specification__genre_1">Adventure</span>
        <span className="specification__genre specification__genre_2">Drama</span>
        <span className="specification__genre specification__genre_3">Family</span>
        <span className="specification__genre specification__genre_4">Fantasy</span>
        <div className="specification__border" />
        <div className="specification__duration">
          <span className="duration__time">1h 46m</span>
        </div>
      </div>
      <Rating />
    </div>
  );
};

export default Description;
