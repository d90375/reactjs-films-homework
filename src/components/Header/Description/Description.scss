.description {
  flex-basis: 70%;
  margin-top: 5rem;
  margin-bottom: 0;

  @media (max-width: 768px) {
    margin-bottom: 1rem;
    margin-top: 1rem;
  }
}

.description__title {
  color: #fff;
  font-family: "HelveticaNeue-bold", serif;
  font-size: 3rem;
  font-weight: 700;
  letter-spacing: 2.4px;
  line-height: 44px;
  margin: 0;
  text-transform: uppercase;

  @media (max-width: 768px) {
    font-size: 1.5rem;
  }
}

.specification__duration {
  color: #fff;
  font-family: "HelveticaNeue-thin", serif;
  font-size: 17px;
  font-weight: 400;
  letter-spacing: 0.96px;
  line-height: 44px;
}

.description__specification {
  display: flex;
  flex-direction: row;
  font-size: 17px;

  @media (max-width: 768px) {
    font-size: 12px;
  }
}

.specification__genre {
  color: #fff;
  font-family: "HelveticaNeue-thin", serif;
  letter-spacing: 0.96px;
  line-height: 44px;
  margin-right: 32px;
}

.specification__border {
  background: #fff;
  height: 1rem;
  margin-right: 32px;
  margin-top: 15px;
  width: 2px;
}
