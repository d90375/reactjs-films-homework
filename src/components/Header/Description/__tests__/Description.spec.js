import React from "react";
import Description from "../Description";

const setUp = (props) => {
  return shallow(<Description {...props} />);
};

describe("Description component", () => {
  let component;
  const mockProps = { genres: "genres", title: "title", voteAverage: 10, runtime: [1, 1] };
  beforeEach(() => {
    component = setUp();
  });

  it("should render Description component without props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render Description component with props", () => {
    component = setUp({ ...mockProps });
    expect(component).toMatchSnapshot();
  });
});
