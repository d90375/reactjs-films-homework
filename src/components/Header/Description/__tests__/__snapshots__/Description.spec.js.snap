// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render Description component with props = (genres: "genres", title: "title", voteAverage: 10, runtime: [1, 1]) 1`] = `
<div
  className="description"
>
  <h2
    className="title"
  >
    title
  </h2>
  <div
    className="specification"
  >
    <span
      className="genre"
    >
      genres
    </span>
    <div
      className="border"
    />
    <div
      className="duration"
    >
      <span>
        1h 1m
      </span>
    </div>
  </div>
  <Rating
    voteAverage={10}
  />
</div>
`;

exports[`Description should render Description component without props 1`] = `
<div
  className="description"
>
  <h2
    className="title"
  />
  <div
    className="specification"
  >
    <span
      className="genre"
    />
    <div
      className="border"
    />
    <div
      className="duration"
    >
      <span>
        0h 0m
      </span>
    </div>
  </div>
  <Rating
    voteAverage={0}
  />
</div>
`;
